streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -693,6 +695,16 @@ public void shouldNotGetQueryMetadataWithPartitionerWhenNotRunningOrRebalancing(
         streams.queryMetadataForKey("store", "key", (topic, key, value, numPartitions) -> 0);
     }
 
+    @Test(expected = UnknownStateStoreException.class)
```

Comment:
I am not sure if using annotation is a good choice. Would it be better to use `assertThrows` ?